capstone-app/src/app/modules/photos/components/image-container/image-container.component.ts

```diff
@@ -52,10 +53,18 @@ export class ImageContainerComponent implements OnChanges, OnInit {
 
   getPhotos() {
     this.originalPhotos = [];
+    const currentQuery = this.query;
     this.photosService.getPhotos(this.query, 10).subscribe(
       (results) => {
+        let isOldRequest = false;
+        if (currentQuery !== this.query) {
+          isOldRequest = true;
+        }
         for (const item in results.items) {
-          if (item) {
+          if (isOldRequest) {
+            break;
+          }
+          else if (item) {
```

Comment:
I should clarify why I made these changes and why they were necessary. Basically, there was a bug that caused the photos of LA to be displayed when the map shows the current location. This was because the app initially started at LA and very quickly switched to current location, so the exports for the current city where very close together in time. Sometimes the getPhotos() request for LA would take longer to complete than the getPhotos request for current location, so the LA photos would be displayed. This change checks to see if the request is an old one and if so, it breaks.